orangecontrib/spectroscopy/widgets/owspectra.py

```diff
@@ -1095,15 +1095,16 @@ def peak_apply(self, position):
             self.plot.addItem(Label_line)
 
     def peak_apply_auto(self, prominence, minHeight, maxHeight, line_overlap):
-        x_axis = getx(self.data)
-        data = np.array(self.data)
+        x_axis = self.data_x
+        data = self.data
```

Comment:
I think the data here is still problematic. See my previous comment. We have to insure, that the code also works in the presence of meta variable and class attributes, this we need to use just X. So it could be `data.X`. But there is an additional problem: wavenumbers might not be ordered. Therefore I suggested `data = data.X[:, self.data_xsind]` in the previous comment. That should be robust.